src/components/base-control.js

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
```

Comment:
Nit: I am not keen on the word `base` in components. To me it emphasizes a programming language aspect rather than something semantic. Would calling it `MapControl` or just `Control` work?